dev/set_matrix.py

```diff
@@ -155,13 +156,28 @@ def read_yaml(location, if_error=None):
         raise
 
 
-def get_released_versions(package_name):
+def days_since_uploaded(dist: dict[str, Any]) -> Optional[int]:
+    """
+    Calculate the number of days since the distribution was uploaded.
+    """
+    if ut := dist.get("upload_time"):
+        return (datetime.now() - datetime.fromisoformat(ut)).days
+    return None
+
+
+def get_released_versions(package_name: str) -> list[Version]:
     data = pypi_json(package_name)
-    versions = []
+    versions: list[Version] = []
     for version, distributions in data["releases"].items():
         if len(distributions) == 0 or any(d.get("yanked", False) for d in distributions):
             continue
 
+        # Ignore versions that were uploaded within the last 2 days to avoid testing unstable
+        # versions. Newly released versions often contain undiscovered bugs
+        # (example: https://github.com/huggingface/transformers/issues/34370).
+        if any(days_since_uploaded(d) < 2 for d in distributions):
```

Comment:
nit: do we want to add an unit test for the case the package is uploaded within last 2 days?

Reply:
I used to write tests for this script, but realized it's questionable if it's worth it. It's a bit tricky to test this script.